### ISSUE TITLE
build with static configuration failed

### ISSUE THREAD

**Issue by markuman** [fluent/fluent-bit#1682]
## Bug Report
using 1.3.2

```
cmake .. -DFLB_HTTP_SERVER=on -DFLB_STATIC_CONF=/home/m/STATIC/
```

succeed with 

```
-- Static configuration path => '/home/m/STATIC/'
-- Processing configuration file: '/home/m/STATIC//fluent-bit.conf'
```

where  `/home/m/STATIC/`

```
 cat /home/m/STATIC/fluent-bit.conf 
[SERVICE]
    Flush     1
    Daemon    off
    Log_Level info
    HTTP_Server  On
    HTTP_Listen  127.0.0.1
    HTTP_PORT    2020

[INPUT]
    Name          exec
    Tag           storage
    Command       printf "{ \"time\": \"%s\", \"value\" : %d }\n" "$(date -u +"%FT%T.000Z")" $(\df --output=avail /mnt |tail -n 1) 
    Interval_Sec  1


[INPUT]
    Name      mem
    Tag       memory
    Interval_Sec  1


[FILTER]
    Name record_modifier
    Match *
    Record hostname ${HOSTNAME}

[OUTPUT]
    Name      stdout
    Match     *
```

running `make` results in 

```
Scanning dependencies of target fluent-bit-static
[ 44%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_config_static.c.o
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c: In function 'flb_config_static_read':
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:177:48: warning: passing argument 3 of 'static_fgets' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  177 |     while (static_fgets(buf, MK_RCONF_KV_SIZE, data, &off)) {
      |                                                ^~~~
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:114:55: note: expected 'char *' but argument is of type 'const char *'
  114 | static int static_fgets(char *out, size_t size, char *data, size_t *off)
      |                                                 ~~~~~~^~~~
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c: At top level:
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:359:18: error: conflicting types for 'flb_config_static_open'
  359 | struct mk_rconf *flb_config_static_open(char *file)
      |                  ^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/include/fluent-bit/flb_worker.h:24,
                 from /home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/include/fluent-bit/flb_log.h:29,
                 from /home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:22:
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/include/fluent-bit/flb_config.h:191:18: note: previous declaration of 'flb_config_static_open' was here
  191 | struct mk_rconf *flb_config_static_open(const char *file);
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c: In function 'flb_config_static_open':
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:369:11: warning: pointer targets in assignment from 'unsigned char *' to 'const char *' differ in signedness [-Wpointer-sign]
  369 |         k = flb_config_files[i][0];
      |           ^
/home/m/git/lekker/fluent-bit/configs/fluent-bit-1.3.2/src/flb_config_static.c:370:11: warning: pointer targets in assignment from 'unsigned char *' to 'const char *' differ in signedness [-Wpointer-sign]
  370 |         v = flb_config_files[i][1];
      |           ^
make[2]: *** [src/CMakeFiles/fluent-bit-static.dir/build.make:765: src/CMakeFiles/fluent-bit-static.dir/flb_config_static.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4916: src/CMakeFiles/fluent-bit-static.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

any ideas?

**Comment by markuman** [comment]
I try older version.  
* 1.2.2 same error
* 1.1.3 it works

to it gets broken after 1.1.3

**Comment by nigels-com** [comment]
I pushed a fix for the compilation errors.  Can you try that branch to see if your issue is resolved?

**Comment by markuman** [comment]
@nigels-com Confirm, pr #1684 fixes the static build.

**Comment by edsiper** [comment]
thanks everyone, this is already fixed in master, it will be backported to 1.3 branch
